bundles/org.openhab.binding.doorbird/src/main/java/org/openhab/binding/doorbird/internal/api/SipStatus.java

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.doorbird.internal.api;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.doorbird.internal.model.SipStatusDTO;
+import org.openhab.binding.doorbird.internal.model.SipStatusDTO.SipStatusBha;
+import org.openhab.binding.doorbird.internal.model.SipStatusDTO.SipStatusBha.SipStatusArray;
+
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link SipStatus} holds SIP status information retrieved from the Doorbell.
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+@NonNullByDefault
+public class SipStatus {
+    private String returnCode = "";
+    private String enable = "";
+    private String prioritizeApp = "";
+    private String registerUrl = "";
+    private String registerUser = "";
+    private String registerPassword = "";
+    private String autocallMotionSensorUrl = "";
+    private String autocallDoorbellUrl = "";
+    private String speakerVolume = "";
+    private String microphoneVolume = "";
+    private String dtmf = "";
+    private String relais1 = "";
+    private String relais2 = "";
+    private String lightPasscode = "";
+    private String incomingCallEnable = "";
+    private String incomingCallUser = "";
+    private String autoNoiseCancellation = "";
+    private String lastErrorCode = "";
+    private String lastErrorText = "";
+    private String ringTimeLimit = "";
+    private String callTimeLimit = "";
+
+    public SipStatus(String sipStatusJson) throws JsonSyntaxException {
+        SipStatusDTO sipStatus = DoorbirdAPI.fromJson(sipStatusJson, SipStatusDTO.class);
+        if (sipStatus != null) {
+            SipStatusBha bha = sipStatus.bha;
+            returnCode = bha.returnCode;
+            for (SipStatusArray sip : bha.sipStatusArray) {
```

Comment:
If I understand this code correctly, you are always overwriting the previous value with either the value of `sip` or the default value. Why not just parse the last array-member? You can then also make all fields final.

Reply:
The returned array contains either zero or 1 element. I don't know why they did it that way. I've adjusted it to just use array index 0, if present. But I don't quite see how I can make the fields final.